### PR TITLE
ug: Re-organize aklite CLI commands documentation

### PR DESCRIPTION
**Pull request by detsch**
# PR Template and Checklist

Please complete as much as possible to speed up the reviewing process.

Readiness and adding reviewers as appropriate is required.

All PRs should be reviewed by a technical writer/documentation team and a peer.
If effecting customers—which is a majority of content changes—a member of Customer Success must also review.

## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

File: source/user-guide/custom-sota-client.rst [hunk 1]
```diff
@@ -194,139 +173,283 @@ the ``--update-name`` option can be used::
...
-
-
 .. warning::
    Downgrading to a older Target is neither recommended or supported by our team;
    doing so may lead to unverified corner cases.
    Only choose to do so mindfully.
    For any update, always test before rolling out to production devices.
 
+When the OSTree image was changed,
+a reboot command is be required after installing the update,
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:192]: `command is be required` - it looks like a typo.

File: source/user-guide/custom-sota-client.rst [hunk 2]
```diff
@@ -194,139 +173,283 @@ the ``--update-name`` option can be used::
...
+- *11*: Failure
+   - Failed to update TUF metadata: TUF metadata is invalid
+- *12*: Failure
+   - Failed to update TUF metadata: TUF metadata is expired
+- *13*: Failure
+   - Failed to update TUF metadata: Download error
+- *14*: Failure
+   - Failed to update TUF metadata: TUF metadata not found in the provided path (offline mode only)
+- *15*: Failure
+   - The bundle metadata is invalid (offline mode only).There are a few reasons why the metadata might be invalid:
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:223]: I would add the reference to the offline update doc https://ci.foundries.io/projects/fio-docs/builds/2867/docs/artifacts/html/user-guide/offline-update/offline-update.html.

File: source/user-guide/custom-sota-client.rst [hunk 3]
```diff
@@ -194,139 +173,283 @@ the ``--update-name`` option can be used::
...
+       1. One or more bundle signatures is/are invalid.
+       2. The bundle Targets` type, whether CI or production, differs from the device`s type.
+       3. The bundle Targets` tag differs from the device`s tag.
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:226]: I would add `4. The offline bundle has expired (its TUF meta has expired)`

File: source/user-guide/custom-sota-client.rst [hunk 4]
```diff
@@ -194,139 +173,283 @@ the ``--update-name`` option can be used::
...
-    - Operation executed successfully
+   - TUF is up to date. No Target update required
+- *3*: Success
+   - Unable to update TUF metadata, using cached metadata
+- *16*: Success
+   - Update is required -- new Target version available
+- *17*: Success
+   - Update is required -- apps need synchronization
+- *18*: Success
+   - Update is required -- rollback to a previous Target
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:294]: How the `check` command can lead to the rollback?

File: source/user-guide/custom-sota-client.rst [hunk 5]
```diff
@@ -462,3 +462,163 @@ while allowing for limited customizations.
...
+                "reason" : "Updating from intel-corei7-64-lmp-93 to intel-corei7-64-lmp-100",
+                "selected" : true,
+                "version" : 100
+        }
+    ]
+
+
+In this scenario, the ``intel-corei7-64-lmp-93`` Target is running.
+Its entry is marked with ``"current" : true``.
+The exit code for the command would be ``16 - Update is required -- new Target version available``,
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:610]: IMO, the update is NOT required in this case, I would say there is/are available updates. The key point is that it is up to a client/user to decide whether an update is required or not. Our utility just informs about available updates.

File: source/user-guide/custom-sota-client.rst [hunk 6]
```diff
@@ -462,3 +462,163 @@ while allowing for limited customizations.
...
+as Target ``intel-corei7-64-lmp-100`` is available.
+If we look into the Target ``intel-corei7-64-lmp-100`` entry, we can notice that is has ``"selected" : true``,
+meaning this is the Target that would be selected by default by ``aktualizr-lite`` to be installed.
+A ``selected`` target always has has a ``reason`` field set as well,
+which describes why this target is supposed to be installed.
+All targets with ``version`` higher than the current one
+are marked as ``"newer": true``.
+
+One additional information that the JSON output presents is if the Target is a failing Target.
+I.e., if the installation of this Target was attempted, but led to a rollback.
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:620]: If there is at least one failing target then it is not correct to write `if the installation of this Target was attempted`. If there is a failing target then there was its installation attempt for sure, not `if`.

File: source/user-guide/custom-sota-client.rst [hunk 7]
```diff
@@ -179,6 +179,15 @@ the ``--update-name`` option can be used::
    Only choose to do so mindfully.
    For any update, always test before rolling out to production devices.
 
+.. note::
+   Since LmP v95, aktualizr-lite won't automatically do a downgrade
```
Comment (mike-sul) [inline source/user-guide/custom-sota-client.rst:183]: I would emphasize that is related to the daemon, e.g. `.. the aktualizr-lite daemon won't`.

Comment (detsch) [reply source/user-guide/custom-sota-client.rst:183]: CLI is affected as well. `update`, `install`, `update` (without explicit version) will not pick an version that is < the current one.
Exit code for `list` and `check` are also impacted.

Comment (mike-sul) [reply source/user-guide/custom-sota-client.rst:183]: OK, then maybe `update` and `install` without parameter - explicit target name/version.

File: source/user-guide/custom-sota-client.rst (hunk 4 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:294]: No rollback is triggered by the `check` command. The command is just telling that an update should be performed, to initiate a rollback. It is related to this case:
    
    // Handle the case when Apps failed to start on boot just after an update.
    // This is only possible with `pacman.create_containers_before_reboot = 0`.

But I'm not 100% sure we can currently hit this situation when calling `list` and `check`.

File: source/user-guide/custom-sota-client.rst (hunk 5 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:610]: It is up to the user to decide what to do with it, but that is the correct return code for the `check` command, which represents the default behavior. If the user is creating a custom logic for this specific part, the exit code can be ignored.

File: source/user-guide/custom-sota-client.rst (hunk 6 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:620]: Fixed (next force push)

File: source/user-guide/custom-sota-client.rst (hunk 5 above)
Comment (mike-sul) [reply source/user-guide/custom-sota-client.rst:610]: It is up to a client to decide on the default behavior.

File: source/user-guide/custom-sota-client.rst (hunk 4 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:294]: Removed exit code 18 from both list and check commands, as we should not get to a situation which this value is returned in the CLI.

File: source/user-guide/custom-sota-client.rst (hunk 2 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:223]: Added

File: source/user-guide/custom-sota-client.rst (hunk 7 above)
Comment (detsch) [reply source/user-guide/custom-sota-client.rst:183]: I've now added an additional phrase to make it more clear

**Review by mike-sul — APPROVED** [review]

**Comment by mike-sul** [comment]
It would be great if @kprosise check this change too.

**Comment by detsch** [comment]
I've replaced a few "target" occurrences with "Target" and force pushed.

@kprosise feel free to merge
